hashicorp/consul-k8s
# Set User-Agent header on Consul API calls

* Upgrade to latest consul api package
* Create common consul package that implements NewClient(). This
  function returns a Consul client that will set the header
  `User-Agent: consul-k8s/<version>` on all calls to Consul.
* Swap to using this new constructor everywhere.
* Add a linter that ensures we use our NewClient() function everywhere
  in the future. Cribbed from https://github.com/hashicorp/lint-consul-retry.

How I've tested this PR:
* unit test
* ran a request printer: `docker run -p 8501:8443 --rm -t mendhak/http-https-echo:17`

  then ran `CONSUL_HTTP_SSL_VERIFY=false consul-k8s get-consul-client-ca -output-file /tmp/out -server-addr localhost -server-port 8501`
  
  ```
  "GET /v1/agent/connect/ca/roots HTTP/1.1" 200 458 "-" "consul-k8s/0.23.0-dev"
  -----------------
  {
      "path": "/v1/agent/connect/ca/roots",
      "headers": {
          "host": "localhost:8501",
          "user-agent": "consul-k8s/0.23.0-dev",
          "accept-encoding": "gzip"
      },
      "method": "GET",
      "body": "",
      "fresh": false,
      "hostname": "localhost",
  ```

How I expect reviewers to test this PR:


Checklist:
